resources/aws_ec2.go

```diff
@@ -146,6 +156,33 @@ type chanStruct struct {
 	err   error
 }
 
+// snsPolicy denotes the structure of sns security policies.
+type snsPolicy struct {
+	Version   string      `json:"Version"`
+	ID        string      `json:"Id"`
+	Statement []statement `json:"Statement"`
+}
+
+// statement denotes the structure of sns security policy statements.
+type statement struct {
+	Sid       string      `json:"Sid"`
+	Effect    string      `json:"Effect"`
+	Principal principal   `json:"Principal"`
+	Action    interface{} `json:"Action"`
+	Resource  string      `json:"Resource"`
+	Condition *struct {
+		StringEquals *struct {
+			AWSSourceOwner *string `json:"AWS:SourceOwner,omitempty"`
+		} `json:"StringEquals,omitempty"`
+	} `json:"Condition,omitempty"`
+}
+
+// principal describes the aws or service account principal.
```

Comment:
does it make sense to call this snsPrincipal to make it more obvious what it's related to?